libraries/AP_HAL_Linux/HAL_Linux_Class.h

```diff
@@ -6,4 +6,11 @@ class HAL_Linux : public AP_HAL::HAL {
 public:
     HAL_Linux();
     void run(int argc, char* const* argv, Callbacks* callbacks) const override;
+
+    void setup_signal_handlers() const;
+
+    static void exit_signal_handler(int);
+
+protected:
+    bool _should_exit = false;
```

Comment:
I didn't want to make it atomic since it will work nonetheless (this or next iteration) and we wouldn't incur in the overhead of loading an atomic on every loop